toolchains/idf/components/toit/CMakeLists.txt

```diff
@@ -0,0 +1,156 @@
+cmake_minimum_required(VERSION 3.11)
+
+set(TOIT_BASE_DIR ${COMPONENT_PATH}/../../../..)
+
+# Register without sources, this will create an INTERFACE lib and we can then specify link options later
+idf_component_register(INCLUDE_DIRS "${TOIT_BASE_DIR}/src"
+        REQUIRES "esp_adc_cal" "mbedtls" "bt" "app_update" "ulp")
+
+cmake_policy(SET CMP0079 NEW)
+
+set(TOIT_SYSTEM_NAME "esp32")
+
+set(TOIT_INTERPRETER_FLAGS "-fno-crossjumping;-fno-tree-tail-merge" CACHE STRING "toit interpreter flags")
+
+set(TOIT_SDK_DIR ${TOIT_BASE_DIR}/build/host/sdk)
+set(TOIT_SDK_DIR ${TOIT_SDK_DIR} PARENT_SCOPE)
+set(TOITC ${TOIT_SDK_DIR}/bin/toit.compile)
+set(TOITPKG ${TOIT_SDK_DIR}/bin/toit.pkg)
+set(TOIT_TOOLS_DIR ${TOIT_SDK_DIR}/tools)
+set(FIRMWARE_BIN ${TOIT_TOOLS_DIR}/firmware)
+set(CMAKE_BUILD_TYPE Release)
+
+add_subdirectory(${TOIT_BASE_DIR} toit EXCLUDE_FROM_ALL)
+
+# Add the include directories from the requires components to the imported toit_vm target
+target_include_directories(toit_vm PRIVATE "$<TARGET_PROPERTY:${COMPONENT_LIB},INTERFACE_INCLUDE_DIRECTORIES>")
+
+# idf cmake does not honor the CMAKE_BUILD_TARGET, so we can not use the CMAKE_C(XX)?_FLAGS.
+# Add the required flags and defines with target_compile_* commands
+target_compile_definitions(toit_vm PRIVATE
+        -DDEPLOY=1
+        -DTOIT_DEPLOY
+        -DESP32
+        -D__FREERTOS__
+        -DRAW)
+
+if (CONFIG_IDF_TARGET_ARCH_XTENSA)
+    set(TOIT_ARCH_COMPILE_OPTIONS -mlongcalls)
+else()
+    set(TOIT_ARCH_COMPILE_OPTIONS -fpermissive -Wno-error=all)
+endif()
+
+target_compile_options(toit_vm PRIVATE
+        -Wno-missing-field-initializers
+        -Wno-ignored-qualifiers
+        -Wno-sign-compare
+        -Wno-extra
+        -Wno-error=format
+        -fjump-tables                   # idf (cmake) sets no-jump-tables which was not set by make
```

Comment:
The interpreter is sitting in IRAM and it has worked with -fjump-tables up to now. An investigation is always a good idea.

Reply:
Erik and I discussed this shortly, and it might be that IRAM only allows certain memory accesses (like alignment...), and that the jump tables could violate that.
Assuming it's an alignment-read issue, then according to Erik: "if it works, it works", and if it ever breaks then the error-message is distinct enough to figure out what's going on.

Reply:
So, let's remove it and add a comment.

Reply:
The jump tables should make things faster, so we should keep it, and just deal with it if we get errors (which should be quite obvious).